### PR TITLE
RFR: Refactor heckell differ

### PR DESCRIPTION
**Pull request by aamontal**
Refactored our implementation of the Heckel Differ to be (hopefully more maintainable).

For the time being I am keeping the original version of the differ around so we can make sure the refactors are sane. 

Also broke up the tests into smaller, files to more easily navigate them. 

Removed some legacy and unused files as well.

### CODE REVIEW

File: lib/dyph3/two_way_differs/heckel_diff.rb [hunk 1]
```diff
@@ -1,85 +1,176 @@
...
 
       private
-        def self.set_frequencies(freq, p, text, flag)
-          text.each_with_index do |line, i|
-            freq[line] ||= {}
-            freq[line][flag] ||= 0
-            freq[line][flag] += 1
-            p[line] = i
+
+        ChangeData = Struct.new(:left_change_pos, :right_change_pos, :change_ranges)
```
Comment (jelder) [inline lib/dyph3/two_way_differs/heckel_diff.rb]: :+1:

**Comment by aamontal** [comment]
@jelder I decided to keep the messy but stable differ around to make sure the new one produces the same results. Planning to update versioned_record and www to run both differs for a time and honeybadger any differences, while still using the stable but messy one.

**Comment by jelder** [comment]
I like it. Think it will have any noticeable performance impact?

Also if you find that there are some areas of intractable complexity, you can silence Rubocop:

https://github.com/bbatsov/rubocop#disabling-cops-within-source-code

**Comment by aamontal** [comment]
@jelder since we are using the original differ for production tasks for now, we could put the comparison into sidekiq and not worry about it.

**Comment by jelder** [comment]
Could be tricky with Dyph3 not currently being aware of Sidekiq.

**Pull request closed without merging**